src/test/java/com/hillel/items_exchange/util/PatternHandlerTest.java

```diff
@@ -0,0 +1,401 @@
+    @MethodSource("createCorrectWordsEmptyOrMin2Max50")
+    void testWordEmptyOrMin2Max50_whenWordIsCorrect(String correctWord) {
+        assertTrue(correctWord.matches(WORD_EMPTY_OR_MIN_2_MAX_50));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createWrongWordsEmptyOrMin2Max50")
+    void testWordEmptyOrMin2Max50_whenWordIsWrong(String wrongWord) {
+        assertFalse(wrongWord.matches(WORD_EMPTY_OR_MIN_2_MAX_50));
+    }
+
+    private static List<String> createCorrectEmails() {
+
+        return List.of(
+                "pushkin@ukr.net",
+                "pushkin.145@gmail.com",
+                "pushkin@mail.ru",
+                "pushkin.pushkin@mail.ru",
+                "pushkin.pushkin@mail.mail.ru",
+                "1@ukr.net",
+                "1@111.ru");
+    }
+
+    private static List<String> createWrongEmails() {
+        List<String> wrongEmails = new ArrayList<>(List.of(
+                "pushkin@ukr.n",
+                ".145@gmail.com",
+                "pushkin@@mail.ru",
+                "1@111.11",
+                "pushkin.11",
+                "pushkin.1u",
+                "pushkin.u1",
+                "1@111.1u",
+                "1@111.1ru",
+                "1@111.ru1",
+                "1@ma~il.ru"));
+
+        List<String> specialCharsAndRusUkrLetters = getSpecialCharacters("\\");
+        specialCharsAndRusUkrLetters.removeAll(List.of("\\.", "\\_", "\\+", "\\-"));
+        specialCharsAndRusUkrLetters.addAll(getRussianAndUkrLetters());
+        specialCharsAndRusUkrLetters.forEach(
+                specialChar -> createCorrectEmails().forEach(
+                        email -> wrongEmails.add(email.replaceAll(
+                                REGROUPED_EMAIL_PATTERN, "$1".concat(specialChar).concat("$2$3$4$5$6"))))
+        );
+
+        return wrongEmails;
+    }
+
+    private static List<String> createCorrectPasswords() {
+        List<String> correctPasswords = new ArrayList<>();
+        String minCorrectPassword = "aA1";
+        correctPasswords.add(minCorrectPassword);
+
+        getSpecialCharacters("\\").forEach(
+                specialChar -> correctPasswords.add(minCorrectPassword.replaceAll(
+                        REGROUPED_PASSWORD_PATTERN, specialChar.concat("$1")))
+        );
+
+        return correctPasswords;
+    }
+
+    private static List<String> createWrongPasswords() {
+        String minCorrectPassword = "aA1";
+        List<String> wrongPasswords = new ArrayList<>(List.of(
+                "",
+                " ",
+                "w",
+                "1",
+                "W",
+                "`"
+        ));
+
+        List<String> tempList = new ArrayList<>();
+        IntStream.range(0, wrongPasswords.size()).forEach(i -> {
+            for (String wrongPassword : wrongPasswords) {
+                tempList.add(wrongPasswords.get(i).concat(wrongPassword));
+                tempList.add(wrongPasswords.get(i).concat(wrongPassword).concat(wrongPasswords.get(i)));
```

Comment:
Could you explain why do you need such code duplication?
`tempList.add(wrongPasswords.get(i).concat(wrongPassword));`